frame/ethereum/src/lib.rs

```diff
@@ -237,7 +236,7 @@ impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
 
 	fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
 		if let Call::transact(transaction) = call {
-			if transaction.signature.chain_id().unwrap_or_default() != T::ChainId::get() {
+			if transaction.signature.chain_id().unwrap_or_default() != pallet_evm::Module::<T>::chain_id() {
```

Comment:
Seems like all the `ChainId` stuff should be moved to `pallet_ethereum`.

Pallet EVM is just a virtual machine. The notion of a chain and a chain id relates to an ethereum-like network.